components/button/index.jsx

```diff
@@ -66,13 +67,10 @@ const Button = createReactClass({
 		/**
 		 * Name of the icon category. Visit <a href="http://www.lightningdesignsystem.com/resources/icons">Lightning Design System Icons</a> to reference icon categories.
 		 */
-		iconCategory: PropTypes.oneOf([
-			'action',
-			'custom',
-			'doctype',
-			'standard',
-			'utility',
-		]),
+		iconCategory: requiredIf(
```

Comment:
There's a default prop set for `iconCategory` so this will really only work (meaning the warning message will only show) if we get rid of the default prop. Do we still want to do this here? Probably a good idea to add `requiredIf` package but maybe this prop doesn't need a conditional given it has a default prop.

Reply:
One option would be to move the default into the component (private), so that it is backwards compatible, but we change the public API so folks get the warning.